pkg/agent/controller/egress/egress_controller.go

```diff
@@ -0,0 +1,628 @@
+package egress
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/interfacestore"
+	"github.com/vmware-tanzu/antrea/pkg/agent/openflow"
+	"github.com/vmware-tanzu/antrea/pkg/apis/egress/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent"
+	"github.com/vmware-tanzu/antrea/pkg/apis/controlplane/v1beta1"
+	opsinformers "github.com/vmware-tanzu/antrea/pkg/client/informers/externalversions/egress/v1alpha1"
+	opslisters "github.com/vmware-tanzu/antrea/pkg/client/listers/egress/v1alpha1"
+	"github.com/vmware-tanzu/antrea/pkg/ovs/ovsconfig"
+)
+
+const (
+	// How long to wait before retrying the processing of a egress policy change.
+	minRetryDelay = 5 * time.Second
+	maxRetryDelay = 300 * time.Second
+	// Default number of workers processing a rule change.
+	defaultWorkers               = 4
+	resyncPeriod   time.Duration = 0
+)
+
+func NewEgressController(
+	kubeClient clientset.Interface,
+	ofClient openflow.Client,
+	ovsBridgeClient ovsconfig.OVSBridgeClient,
+	nodeIP net.IP,
+	egressInformer opsinformers.EgressInformer,
+	antreaClientGetter agent.AntreaClientProvider,
+	ifaceStore interfacestore.InterfaceStore,
+) *Controller {
+	c := &Controller{
+		kubeClient:           kubeClient,
+		nodeIP:               nodeIP,
```

Comment:
The SNAT IP can be any IP configured on the node, not only the Node IP in K8s Node IP.
If you are not sure how to get it, I could provide an interface like `LocalIPDetector.IsLocalIP(ip string) bool` for you to consume.

Reply:
Thanks, I will try to implement `LocalIPDetector.IsLocalIP(ip string) bool` by myself first.

Reply:
I have added `IsLocalIP(ip string)` in this PR. Please check if it is correct, thanks